### ISSUE TITLE
PHP 5.4 and higher or maybe 5.6 or higher

### ISSUE THREAD

**Issue by jsteggink** [solariumphp/solarium#489]
Hi,

How ready is the Solarium community for PHP 5.4 or higher? In order to modularize Solarium more, it would be great to use Traits to split up parts of the code. 

The minimum version of PHP is currently set to 5.3.2 (in composer.json) and higher, even though it says 5.4 in the README. Can we change 5.3.2 to 5.4, or maybe even 5.6? Everything before 5.6 is end-of-life anyways.

I'm interested in your views on this.

Cheers,
Jeroen

**Comment by chadicus** [comment]
I think the next major version of Solarium (v4) should be php 7 only.

**Comment by jsteggink** [comment]
I'm working on an extension for solarium to be able to use SolrCloud, called solarium-cloud (#478). While working on it I encountered numerous difficulties. Some of the classes are too rigid and coupled. Furthermore, some of the code is a little bit outdated. Nothing wrong with it, but with PHP 7 we can make it a bit .

I decided to write the extension in PHP 7, since the future users I have in mind are early adopters. One of the goals is to implement more of Solr's features to the PHP community.
